lib/scholar/linear_regression.ex

```diff
@@ -0,0 +1,42 @@
+defmodule Scholar.LinearRegression do
+
+  defstruct :coefficients
+
+  @doc """
+  Fits a linear regression model to the given dataset.
+  """
+  def fit(x, y) do
```

Comment:
I like `fit` better than `train` because it's more general and can encompass cases such as this one where there's no iterative learning

Reply:
Agreed @polvalente!